### PR TITLE
Document how to specify a static resource bundle to load as the last …

### PR DESCRIPTION
**Pull request by stevepiercy**
…one in Classic UI

See:
- https://github.com/plone/Products.CMFPlone/pull/4054
- https://github.com/plone/Products.CMFPlone/pull/4076 👈 this one must be merged before this PR can be merged.
- https://github.com/plone/Products.CMFPlone/pull/4077 👈 this one must be merged before this PR can be merged.
- https://github.com/plone/plone.app.theming/pull/248

----
📚 Documentation preview 📚: https://plone6--1802.org.readthedocs.build/

### CODE REVIEW

**Comment by stevepiercy** [comment]
Direct link to preview of docs changes at https://plone6--1802.org.readthedocs.build/classic-ui/static-resources.html#loading-order-of-resources

File: docs/classic-ui/static-resources.md [hunk 1]
```diff
@@ -100,4 +100,20 @@ The following attributes are available for registering a static resource:
 
 ## Loading order of resources
 
-`depends` is used to define the loading order of resources by specifying the name of the depending bundle.
+You can use the `depends` attribute to define the order in which resources load.
+
+You can specify a list of either the names of the bundles on which this bundle depends or a single list item of `["all"]`.
+In the latter case of `["all"]` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.
```
Comment (jensens) [inline docs/classic-ui/static-resources.md]: What happens if there are two bundles depending on "all"? Is this prevented? If not, is there some order or is it random?

Comment (petschki) [reply docs/classic-ui/static-resources.md]: I'd say then the alphabetical sorting in the resource registry takes effect, which you could manipulate with special prefixes (eg. "zzz_mybundle")

Comment (thet) [reply docs/classic-ui/static-resources.md]: ~~If more bundles depend on "all", all the "all" bundles are rendered in the order they were defined~~

That was apparently wrong. @petschki answer above is correct.

File: docs/classic-ui/static-resources.md [hunk 2]
```diff
@@ -100,4 +100,20 @@ The following attributes are available for registering a static resource:
...
+
+This feature lets you override a theme with custom CSS from a bundle.
+Previously you had to add the CSS customizations to the registry via the `custom_css` settings, then update the registry after every change.
+
+```{versionadded} Plone 6.0.14
+The `depends` attribute may be assigned the value of `["all"]`, making this bundle render last, after all other bundles.
+```
+
+```{note}
+Setting `depends=["all"]` does not affect custom CSS that you define in the {guilabel}`Theming` control panel, which _always_ renders as the last style resource.
```
Comment (thet) [inline docs/classic-ui/static-resources.md]:
```suggestion
Setting `depends="all"` does not affect custom CSS that you define in the {guilabel}`Theming` control panel, which _always_ renders as the last style resource.
```

File: docs/classic-ui/static-resources.md [hunk 3]
```diff
@@ -100,4 +100,20 @@ The following attributes are available for registering a static resource:
...
+
+You can specify a list of either the names of the bundles on which this bundle depends or a single list item of `["all"]`.
+In the latter case of `["all"]` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.
+This includes the theme, such as Barceloneta's theme CSS.
+
+This feature lets you override a theme with custom CSS from a bundle.
+Previously you had to add the CSS customizations to the registry via the `custom_css` settings, then update the registry after every change.
+
+```{versionadded} Plone 6.0.14
+The `depends` attribute may be assigned the value of `["all"]`, making this bundle render last, after all other bundles.
```
Comment (thet) [inline docs/classic-ui/static-resources.md]:
```suggestion
The `depends` attribute may be assigned the value of `"all"`, making this bundle render last, after all other bundles.
```

File: docs/classic-ui/static-resources.md [hunk 4]
```diff
@@ -100,4 +100,20 @@ The following attributes are available for registering a static resource:
...
 
-`depends` is used to define the loading order of resources by specifying the name of the depending bundle.
+You can use the `depends` attribute to define the order in which resources load.
+
+You can specify a list of either the names of the bundles on which this bundle depends or a single list item of `["all"]`.
+In the latter case of `["all"]` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.
+This includes the theme, such as Barceloneta's theme CSS.
+
+This feature lets you override a theme with custom CSS from a bundle.
+Previously you had to add the CSS customizations to the registry via the `custom_css` settings, then update the registry after every change.
```
Comment (thet) [inline docs/classic-ui/static-resources.md]:
```suggestion
Note: You can also add CSS customizations through-the-web via the theming control panel.
There you can add some custom CSS.
These custom CSS file is always rendered last, also after any bundles with a dependency of "all"
```

File: docs/classic-ui/static-resources.md (hunk 1 above)
Comment (thet) [inline docs/classic-ui/static-resources.md]:
```suggestion
You can specify a comma separated list of either the names of the bundles on which this bundle depends or the special value `"all"`.
In the latter case of `"all"` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.
If multiple bundles use the `"all"` dependency, they are all output at the end, in the order they were defined.
```

**Review by thet — DISMISSED** [review]
Approved with my suggestions.

File: docs/classic-ui/static-resources.md (hunk 1 above)
Comment (stevepiercy) [reply docs/classic-ui/static-resources.md]: Addressed in f80c6103029f37c3e9bfb9c5fc70292f36442881

File: docs/classic-ui/static-resources.md (hunk 2 above)
Comment (stevepiercy) [reply docs/classic-ui/static-resources.md]: I realized that we use XML syntax in the examples, so I changed these to be plain old inline literals.

File: docs/classic-ui/static-resources.md (hunk 4 above)
Comment (stevepiercy) [reply docs/classic-ui/static-resources.md]: I merged content and rewrote this one.

File: docs/classic-ui/static-resources.md (hunk 1 above)
Comment (stevepiercy) [reply docs/classic-ui/static-resources.md]: This content was redundant after merging your content, so I removed it.

**Review by jensens — DISMISSED** [review]

File: docs/classic-ui/static-resources.md [hunk 5]
```diff
@@ -72,32 +76,52 @@ Registering a JavaScript file and a CSS file in the same bundle is also possible
...
-`load_defer`
-:   Whether the bundle should be loaded deferred or not.
-    *Only JavaScript*
+    If the bundle's dependencies do not exist, then the bundle will not render.
 
+    The `depends` attribute may be assigned the value of `all`, making this bundle render last, after all other bundles.
+    The `all` value lets you load CSS files after the automatically added theme resources and override CSS declarations from your own custom CSS files.
 
-(classic-ui-static-resources-loading-order-label)=
+    If you set multiple bundles to `all`, then these bundles will render in the order in which you define them.
```
Comment (petschki) [inline docs/classic-ui/static-resources.md]:
```suggestion
    If you set multiple bundles to `all`, then these bundles will render in alphabetical order by its name.
```

**Review by petschki — CHANGES_REQUESTED** [review]

File: docs/classic-ui/static-resources.md [hunk 6]
```diff
@@ -52,15 +56,15 @@ You can register a CSS resource in the same way.
   </registry>
 ```
 
-Registering a JavaScript file and a CSS file in the same bundle is also possible.
+You can also register a JavaScript file and a CSS file in the same bundle.
```
Comment (yurj) [inline docs/classic-ui/static-resources.md]: "It is also possible to include both a JavaScript file and a CSS file in the same bundle." If you don't want to use "you".
```suggestion
It is also possible to include both a JavaScript file and a CSS file in the same bundle.
```

**Review by thet — DISMISSED** [review]

File: docs/classic-ui/static-resources.md (hunk 6 above)
Comment (stevepiercy) [reply docs/classic-ui/static-resources.md]: I want to address the reader with the active voice of "you can", not the passive voice of "it is possible". The former tone is more encouraging and friendly to the reader. It's a thing we adopted from the [Microsoft Writing Style Guide](https://learn.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice), as noted in https://6.docs.plone.org/contributing/documentation/authors.html#plone-documentation-styleguide.

**Review by petschki — DISMISSED** [review]

**Comment by stevepiercy** [comment]
Thanks everyone for your collaboration!

**Pull request merged**